PIMbench/cpp-relu/PIM/relu.cpp

```diff
@@ -259,72 +254,74 @@ int main(int argc, char *argv[]) {
   int inputWidth = inputMatrix[0][0].size();
 
   // Define parameters for processing
-  unsigned numCols = 8192, numOfCore = 4096;
-  int numOfPIMRow = 1;
-  int numOfPIMColumn = (inputHeight * inputWidth / numOfPIMRow);
-  int numOfMatPerRow = std::min(static_cast<int>(std::floor((1.0 * numCols * numOfCore) / numOfPIMColumn)), inputDepth);
+  PimDeviceProperties deviceProp;
+  PimStatus status = pimGetDeviceProperties(&deviceProp);
+  if (status != PIM_OK) {
+    std::cout << "Abort: pimGetDeviceProperties failed" << std::endl;
+    return 1;
+  }
+  // Get the device parameters
+  uint64_t numCols = deviceProp.numColPerSubarray;
+  uint64_t numRows = deviceProp.numRowPerSubarray;
+  uint64_t numOfBits = uint64_t(deviceProp.numRanks) * uint64_t(deviceProp.numBankPerRank) * uint64_t(deviceProp.numSubarrayPerBank) * numCols * numRows;
+
+  uint64_t numOfPIMRow = 1;
```

Comment:
Why is this value 1?

Reply:
'1' here indicates that ReLU operation is an element-wise operation. So, for each element, we do max(0, element) operation.  Hence, the variable is assigned a value of 1.